sql-parser/src/main/java/io/crate/sql/parser/Statement.g

```diff
@@ -861,6 +868,15 @@ createRepositoryStmt
       (WITH '(' genericProperties ')' )? -> ^(CREATE_REPOSITORY repository ident genericProperties?)
     ;
 
+createSnapshotStmt
+    : qname
+      ALL
+      (WITH '(' genericProperties ')' )? -> ^(CREATE_SNAPSHOT qname ALL genericProperties?)
+    | qname
```

Comment:
you could create a separate rule for  ``(ALL | TABLE tableWithPartitionList)`` and use its result in the tree created here, so you don't need to repeat the WITH clause